### PR TITLE
Cherry-picks for the 4.2 branch (future 4.2.2) - 3rd batch

### PR DESCRIPTION
**Pull request by akien-mga**
Long time no see `4.2`...

WIP, I'll add more commits.

For now this batch includes:
- Buildsystem changes
- Thirdparty changes
- Some high priority bug fixes and regression fixes
- Safe-looking rendering fixes
- Documentation improvements
- Editor fixes
- Physics fixes
- Platform fixes
- Core fixes and more, the rest of the "safe" looking stuff

### CODE REVIEW

File: editor/import/resource_importer_obj.cpp [hunk 1]
```diff
@@ -411,6 +411,19 @@ static Error _parse_obj(const String &p_path, List<Ref<Mesh>> &r_meshes, bool p_
 
 				mesh = surf_tool->commit(mesh, mesh_flags);
 
+				if (mesh_flags & RS::ARRAY_FLAG_COMPRESS_ATTRIBUTES && generate_tangents) {
+					// Compression is enabled, so let's validate that the normals and tangents are correct.
+					Vector<Vector3> norms = array[Mesh::ARRAY_NORMAL];
```
Comment (AThousandShips) [inline editor/import/resource_importer_obj.cpp:416]: These changes depend on:
* https://github.com/godotengine/godot/pull/86365

It seems

Comment (akien-mga) [reply editor/import/resource_importer_obj.cpp:416]: Right, that's something I wanted to check with @clayjohn on Monday.

**Comment by akien-mga** [comment]
Added a few `topic:rendering` cherrypicks. The rest need to be discussed with Clay: https://github.com/godotengine/godot/pulls?q=is%3Apr+label%3Acherrypick%3A4.2+is%3Aclosed+label%3Atopic%3Arendering

File: editor/import/resource_importer_obj.cpp (hunk 1 above)
Comment (akien-mga) [reply editor/import/resource_importer_obj.cpp:421]: I included that PR for now as it seemed harmless enough.

**Comment by akien-mga** [comment]
Added a batch with all the documentation cherry-picks.

**Comment by akien-mga** [comment]
Added a batch with a handful of editor fixes (did not include all of them, some looked riskier or didn't cherry-pick cleanly).

**Comment by akien-mga** [comment]
And added a batch with platform specific bugfixes.

**Comment by akien-mga** [comment]
This is now ready for review.

I'd welcome some help to check the list of commits I cherry-picked, and flag any that seems potentially risky / best to drop for now and not cherry-pick to 4.2.x or consider for a later build.

**Comment by Mickeon** [comment]
Nitpick but if Transform3D class reference overhaul is in, so should https://github.com/godotengine/godot/pull/87175 and https://github.com/godotengine/godot/pull/87181 as they heavily borrow from one another.

**Comment by clayjohn** [comment]
> Added a few `topic:rendering` cherrypicks. The rest need to be discussed with Clay: https://github.com/godotengine/godot/pulls?q=is%3Apr+label%3Acherrypick%3A4.2+is%3Aclosed+label%3Atopic%3Arendering

I removed the label from the PRs I don't think should be cherrypicked. The remaining ones are safe to include if you want

**Comment by clayjohn** [comment]
I don't see anything on the list that needs to be removed! Great work

**Comment by Sauermann** [comment]
I had a look at PRs, that I was involved in and are looking good.

**Comment by tokengamedev** [comment]
I could not find https://github.com/godotengine/godot/pull/87346 in the list. It is critical to have this change.

**Comment by akien-mga** [comment]
> I could not find #87346 in the list. It is critical to have this change.

I need to assess the pros and cons with @m4gr3d. I was earmarking it for 4.2.3 and trying not to increase the scope _too much_ for 4.2.2, but this batch makes it somewhat big already so we might want to go all the way.

**Comment by tokengamedev** [comment]
I appreciate, reviewing again. I have been trying to build plugins on top of 4.2.2 RC and if merged let me know I will do some testing for android export as well as plugin creations. If anything, else needs to be tested also let me know.

**Comment by akien-mga** [comment]
> > I could not find #87346 in the list. It is critical to have this change.
> 
> I need to assess the pros and cons with @m4gr3d. I was earmarking it for 4.2.3 and trying not to increase the scope _too much_ for 4.2.2, but this batch makes it somewhat big already so we might want to go all the way.

Let's give it a go, and see how it fares under testing :)

**Comment by tokengamedev** [comment]
Awesome!! thanks

**Comment by tokengamedev** [comment]
I didn't know where to mention this, but initial tests on https://github.com/godotengine/godot/pull/87346 shows it is working fine.

**Pull request merged**